app/models/importers/hmis_six_one_one/base.rb

```diff
@@ -91,16 +91,35 @@ def import!
           @import.save
           import_services()
 
+          delete_remaining_pending_deletes()
           complete_import()
           match_clients()
           project_cleanup()
           log("Import complete")
         ensure
+          cleanup_any_pending_deletes()
           remove_import_files() if @remove_files
         end
       end # end with_advisory_lock
     end
 
+    def delete_remaining_pending_deletes()
+      # If a pending delete is still present, the associated record is not in the import, and should be
+      # marked as deleted
+      soft_deletable_sources.each do |source|
```

Comment:
We should be able to do this completely in sql
`update table_name set DateDeleted = pending_date_deleted where pending_date_deleted is not null`  (in Arel of course)

Reply:
`source.where.not(pending_date_deleted: :nil).update_all(DateDeleted: source.arel_table[:pending_date_deleted], pending_date_deleted: nil)` should be pretty close.